src/shop/tests/test_api.py

```diff
@@ -27,9 +32,13 @@ def parse_delivery_report(self, report_path):
     def retrieve_product(self, sku):
         return SupplierProduct(
             name='Billys Original',
-            price=Decimal('9.25')
+            price=Decimal('9.25'),
+            units=1
         )
 
+    def order_product(self, sku, qty):
+        pass
```

Comment:
It's just something we need to do as long we keep `DummySupplierAPI` around. All the methods in `SupplierBase` are decorated with `@abstractmethod` and all the children classes need to implement all those methods, otherwise hell will break loose and Python won't run your code.